test/integration/apiserver/apply/benchmark_test.go

```diff
@@ -0,0 +1,385 @@
+        secretKeyRef:
+          key: other-key
+          name: other-name
+    - name: VAR_1
+      valueFrom:
+        secretKeyRef:
+          key: some-key
+          name: some-name
+    image: some-image-name
+    imagePullPolicy: IfNotPresent
+    name: some-name
+    resources:
+      requests:
+        cpu: "0"
+    terminationMessagePath: /dev/termination-log
+    terminationMessagePolicy: File
+    volumeMounts:
+    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
+      name: default-token-hu5jz
+      readOnly: true
+  dnsPolicy: ClusterFirst
+  nodeName: node-name
+  priority: 0
+  restartPolicy: Always
+  schedulerName: default-scheduler
+  securityContext: {}
+  serviceAccount: default
+  serviceAccountName: default
+  terminationGracePeriodSeconds: 30
+  tolerations:
+  - effect: NoExecute
+    key: node.kubernetes.io/not-ready
+    operator: Exists
+    tolerationSeconds: 300
+  - effect: NoExecute
+    key: node.kubernetes.io/unreachable
+    operator: Exists
+    tolerationSeconds: 300
+  volumes:
+  - name: default-token-hu5jz
+    secret:
+      defaultMode: 420
+      secretName: default-token-hu5jz
+status:
+  conditions:
+  - lastProbeTime: null
+    lastTransitionTime: "2019-07-08T09:31:18Z"
+    status: "True"
+    type: Initialized
+  - lastProbeTime: null
+    lastTransitionTime: "2019-07-08T09:41:59Z"
+    status: "True"
+    type: Ready
+  - lastProbeTime: null
+    lastTransitionTime: null
+    status: "True"
+    type: ContainersReady
+  - lastProbeTime: null
+    lastTransitionTime: "2019-07-08T09:31:18Z"
+    status: "True"
+    type: PodScheduled
+  containerStatuses:
+  - containerID: docker://885e82a1ed0b7356541bb410a0126921ac42439607c09875cd8097dd5d7b5376
+    image: some-image-name
+    imageID: docker-pullable://some-image-id
+    lastState:
+      terminated:
+        containerID: docker://d57290f9e00fad626b20d2dd87a3cf69bbc22edae07985374f86a8b2b4e39565
+        exitCode: 255
+        finishedAt: "2019-07-08T09:39:09Z"
+        reason: Error
+        startedAt: "2019-07-08T09:38:54Z"
+    name: name
+    ready: true
+    restartCount: 6
+    state:
+      running:
+        startedAt: "2019-07-08T09:41:59Z"
+  hostIP: 10.0.0.1
+  phase: Running
+  podIP: 10.0.0.1
+  qosClass: BestEffort
+  startTime: "2019-07-08T09:31:18Z"
+`)
+
+func decodePod(podBytes []byte) v1.Pod {
+	pod := v1.Pod{}
+	err := yaml.Unmarshal(podBytes, &pod)
+	if err != nil {
+		panic(err)
+	}
+	return pod
+}
+
+func encodePod(pod v1.Pod) []byte {
+	podBytes, err := yaml.Marshal(pod)
+	if err != nil {
+		panic(err)
+	}
+	return podBytes
+}
+
+func BenchmarkNoServerSideApply(b *testing.B) {
+	defer featuregatetesting.SetFeatureGateDuringTest(b, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, false)()
+
+	_, client, closeFn := setup(b)
+	defer closeFn()
+	flag.Lookup("v").Value.Set("0")
+
+	benchAll(b, client, decodePod(podBytes))
+}
+
+func getPodSizeWhenEnabled(b *testing.B, pod v1.Pod) int {
```

Comment:
passing bigger objects like v1.Pod by pointer is preferred to avoid the copy by value on each invocation.

Reply:
Just moving the test...

Reply:
right, it's fine to keep as is IMO.